### PR TITLE
fix: assign cache value for default configs

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

fix

**Did you add tests for your changes?**
WIP

**If relevant, did you update the documentation?**
NA

**Summary**
We assign config path in cache properties passed to the compiler, it didn't take into account the default configs which are resolved so this should fix it.

**Does this PR introduce a breaking change?**
Nope

**Other information**
Tests WIP

### CODE REVIEW

File: packages/webpack-cli/lib/groups/ConfigGroup.js [hunk 1]
```diff
@@ -126,21 +126,21 @@ const resolveConfigFiles = async (args) => {
...
+        opts = await finalize(foundConfig, args, true);
 
         return;
     }
 };
 
 // Given config data, determines the type of config and
 // returns final config
-const finalize = async (moduleObj, args) => {
+const finalize = async (moduleObj, args, isDefault = false) => {
```
Comment (snitin315) [inline packages/webpack-cli/lib/groups/ConfigGroup.js:143]:
```suggestion
const finalize = async (moduleObj, args, isDefaultConfig = false) => {
```

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:143]: after ci

**Comment by anshumanv** [comment]
@evilebottnawi random failures 😞

**Comment by evilebottnawi** [comment]
@anshumanv Let's do rebase

**Comment by evilebottnawi** [comment]
I think we have a bug in implementation, because:
```
  buildDependencies: {
        defaultWebpack: [
          '/home/runner/work/webpack-cli/webpack-cli/node_modules/webpack/lib/'
        ]
      },
```

File: test/core-flags/cache-flags.test.js [hunk 2]
```diff
@@ -136,6 +136,17 @@ describe('cache related flags from core', () => {
         expect(exitCode).toEqual(0);
     });
 
+    it('should assign cache build dependencies with default config', () => {
+        // TODO: Fix on windows
+        if (isWindows) return;
```
Comment (evilebottnawi) [inline test/core-flags/cache-flags.test.js:141]: Avoid this here

Comment (anshumanv) [reply test/core-flags/cache-flags.test.js:141]: already skipped for windows, some problems in all tests 😞

**Comment by anshumanv** [comment]
> I think we have a bug in implementation, because:

This doesn't happen in all tests, which is very strange 😞

**Comment by evilebottnawi** [comment]
Can you try it locally?

**Comment by anshumanv** [comment]
passing locally 😞

**Comment by evilebottnawi** [comment]
Can you try to use `node_modules/bin/jest test/path/to/name.test.js` for this test (change it in github actions file), so we will run the only related tests

File: test/core-flags/cache-flags.test.js (hunk 2 above)
Comment (evilebottnawi) [reply test/core-flags/cache-flags.test.js:141]: Yep, let's focus on it late

File: packages/webpack-cli/lib/groups/ConfigGroup.js (hunk 1 above)
Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/ConfigGroup.js]: @anshumanv Can you fix it, I want to merge it and do release

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/ConfigGroup.js]: fixed

**Review by evilebottnawi — APPROVED** [review]
Good, wait CI green, they are very slow :disappointed:

**Comment by evilebottnawi** [comment]
macos is broken today, but should be fine, because linux is green, let's merge and do release

**Pull request merged**